src/ts/component/page/main/date.tsx

```diff
@@ -0,0 +1,301 @@
+import * as React from 'react';
+import $ from 'jquery';
+import raf from 'raf';
+import { observer } from 'mobx-react';
+import { Header, Footer, Loader, Deleted, ListObject, Button, Icon } from 'Component';
+import { I, M, C, S, U, J, Action, keyboard, translate } from 'Lib';
+import Controls from 'Component/page/elements/head/controls';
+import HeadSimple from 'Component/page/elements/head/simple';
+
+interface State {
+	isLoading: boolean;
+	isDeleted: boolean;
+	relations: Item[];
+	selectedRelation: string;
+};
+
+interface Item {
+	relationKey: string;
+	counter: number;
+}
+
+const SUB_ID = 'dateListObject';
+
+const PageMainDate = observer(class PageMainDate extends React.Component<I.PageComponent, State> {
+
+	_isMounted = false;
+	node: any = null;
+	id = '';
+	refHeader: any = null;
+	refHead: any = null;
+	refList: any = null;
+	refCalIcon: any = null;
+	refControls: any = null;
+	loading = false;
+	timeout = 0;
+
+	state:State = {
+		isLoading: false,
+		isDeleted: false,
+		relations: [],
+		selectedRelation: 'createdDate',
+	};
+
+	constructor (props: I.PageComponent) {
+		super(props);
+		
+		this.resize = this.resize.bind(this);
+	};
+
+	render () {
+		const { space } = S.Common;
+		const { isLoading, isDeleted, relations, selectedRelation } = this.state;
+		const rootId = this.getRootId();
+		const check = U.Data.checkDetails(rootId);
+		const object = S.Detail.get(rootId, rootId, ['timestamp']);
+
+		if (isDeleted) {
+			return <Deleted {...this.props} />;
+		};
+
+		const creatorRelation = S.Record.getRelationByKey('creator');
+		const columns: any[] = [
+
+			{ relationKey: 'type', name: translate('commonObjectType'), isObject: true },
+			{ 
+				relationKey: creatorRelation.relationKey, name: creatorRelation.name, isObject: true,
+			},
+		];
+		// TODO: add calendar 00:49:16
+		// TODO: dark theme
+		// TODO: create task for middle to return relations in the order corresponding to UI design
+		const filters: I.Filter[] = [
+			{ 
+				relationKey: selectedRelation, 
+				condition: I.FilterCondition.Equal, 
+				value: object.timestamp, 
+				format: I.RelationType.Date 
+			},
+		];
+
+		let content = null;
+
+		if (isLoading) {
+			content = <Loader id="loader" />;
+		} else {
+			const calendarMenu = <><Icon className="arrow left"/><Icon className="arrow right"/><Icon ref={ref => this.refCalIcon = ref} className="calendar" onClick={this.onCalendar}/></>;
+
+			content = (
+				<div className="blocks wrapper">
+						<Controls ref={ref => this.refControls = ref} key="editorControls" {...this.props} rootId={rootId} resize={this.resize} />
```

Comment:
Controls are irrelevant here, should be removed and need to add Header as in other objects.